docs/getting-started-guides/scratch.md

```diff
@@ -0,0 +1,664 @@
+  - Build and push your own images.
+    - Necessary if you are using a private registry.
+    - The release contains files such as `./kubernetes/server/bin/kube-apiserver.tar` which
+      can be converted into docker images using a command like
+      `tar -C kube-apiserver -c . | docker import - kube-apiserver`
+    - *TODO*: test above command.
+
+You may also run etcd in a pod, in which case you can:
+  - Use a GCR image, such as `gcr.io/google_containers/etcd:2.0.12`
+  - Build your own image.
+  - Build your own image using `kubernetes/cluster/images/etcd/Dockerfile` and
+   `kubernetes/cluster/images/etcd/Makefile`
+
+The remainder of the document assumes that the image identifiers have been chosen and stored in corresponding env vars.  Examples (replace with latest tags and adjust if using private registry):
+  - `APISERVER_IMAGE=gcr.io/google_containers/kube-apiserver:$TAG`
+  - `SCHEDULER_IMAGE=gcr.io/google_containers/kube-scheduler:$TAG`
+  - `CNTRLMNGR_IMAGE=gcr.io/google_containers/kube-controller-manager:$TAG`
+
+### Configuring Base Node Software
+This section discusses how to configure machines to be kuberetes nodes. 
+
+You should run three daemons on every node:
+  - docker or rkt
+  - kubelet
+  - kube-proxy
+
+You will also need to do assorted other configuration on top of a
+base OS install.
+
+One possible starting point is to setup a cluster using an existing Getting
+Started Guide and then copy the init.d scripts or systemd unit files from that
+cluster, and then modify them for use on your custom cluster.
+
+#### Configuring docker
+All nodes should have docker installed.
+  - Minimum required docker or rkt version will vary as Kubernetes version changes.
+  - Newest stable is a good choice.
+  - Kubelet will log a warning and refuse to start pods if the version is too old, so try it.
+
+The way you configure docker will depend in whether you have chosen the routable-vip or overlay-network approaches for your network.
+Some docker options will want to think about:
+  - `--bridge cbr0`
+  - `--iptables=false`
+  - `--ip-masq=false`
+    - if have setup PodIPs to be routable, then you want this false, otherwise, docker will
+      rewrite the PodIP source-address to a NodeIP, to allow the Pod to talk to IPs outside the
+      cluster.
+    - if you are using an overlay network, consult those instructions.  
+  - `--bip=` and `--mtu=`
+    - may be required when using Flannel; see Flannel instructions.
+  - `--insecure-registry $CLUSTER_SUBNET` 
+    - to connect to a private registry, if you set one up, without using SSL.
+
+You may want to increase the number of open files for docker:
+   - `DOCKER_NOFILE=1000000`
+
+Ensure docker is working correctly on your system before proceeding with the rest of the
+installtion, by following examples given in the Docker documentation.
+
+Remember to eliminate any network configuration created as part of running docker previously
+with different options:
```

Comment:
Is this run once (after its "working correctly") or run after each set of Docker options that you try?

 - For run once at the end (after you get it working), revise sentence to:
"To eliminate any network configuration created from previously running docker with different options, run the following commands:"
 - For run after each set of Docker options:
     - We should move this sentence up above the sentence at line 281 ("Ensure Docker is working....")
     - Revise to:
"Remember that for each set of Docker options you try, you have to first clean up the network configuration before trying different options. To remove the previous network configuration, run the following commands:"